bech32/app/Main.hs

```diff
@@ -100,19 +109,20 @@ parse = customExecParser (prefs showHelpOnEmpty) parser
         ]
 
     cmd :: Parser Cmd
-    cmd = Cmd <$> optional hrpArgument
+    cmd = (RunCmd <$> optional hrpArgument) <|> (VersionCmd <$ versionSwitch)
+
+versionSwitch :: Parser Bool
```

Comment:
I think that `--version` is actually a flag not a switch. That is, this type should be `Parser ()`, or perhaps `Parser Cmd`.